File: src/main/java/com/qtech/bubbles/common/logging/GameLogFormatter.java
package com.qtech.bubbles.common.logging;

import org.fusesource.jansi.Ansi;

import java.io.PrintWriter;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.TimeZone;

public class GameLogFormatter {
    public void publish(GameLogRecord record, PrintWriter printer) {
        Date date = new Date(record.getMilliTime());
        DateFormat formatter = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss.SSS");
        formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
        String dateFormatted = formatter.format(date);

        SimpleDateFormat.getDateTimeInstance();

        record.getLevel().getStream().println(Ansi.ansi().fg(record.getLevel().getAnsiColor()).a(dateFormatted + " | " + record.getLoggerName() + " | " + record.getLevel().getName().toUpperCase() + ": " + record.getMessage()).reset());
        record.getLevel().getStream().flush();
        printer.println(dateFormatted + " | " + record.getLoggerName() + " | " + record.getLevel().getName().toUpperCase() + ": " + record.getMessage());
        printer.flush();
    }
}

File: src/main/kotlin/qtech/bubbles/common/logging/GameLogFormatter.kt
package qtech.bubbles.common.logging

import org.fusesource.jansi.Ansi
import java.io.PrintWriter
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

class GameLogFormatter {
    fun publish(record: GameLogRecord, printer: PrintWriter?) {
        val date = Date(record.milliTime)
        val formatter: DateFormat = SimpleDateFormat("dd-MM-yyyy HH:mm:ss.SSS")
        formatter.timeZone = TimeZone.getTimeZone("UTC")
        val dateFormatted = formatter.format(date)
        SimpleDateFormat.getDateTimeInstance()
        record.level.stream.println(Ansi.ansi().fg(record.level.ansiColor).a(dateFormatted + " | " + record.loggerName + " | " + record.level.text.uppercase(Locale.getDefault()) + ": " + record.message).reset())
        record.level.stream.flush()
        printer!!.println(dateFormatted + " | " + record.loggerName + " | " + record.level.text.uppercase(Locale.getDefault()) + ": " + record.message)
        printer.flush()
    }
}